src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1644,76 +1644,105 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
+            ref ushort source = ref Unsafe.As<char, ushort>(ref _firstChar);
 
-                while (offset <= lengthToExamine - (nuint)Vector256<ushort>.Count)
+            if (Vector256.IsHardwareAccelerated)
+            {
+                if (lengthToExamine >= (nuint)Vector256<ushort>.Count)
                 {
-                    Vector256<ushort> vector = Vector256.LoadUnsafe(ref source, offset);
-                    Vector256<ushort> vector1 = Vector256.Equals(vector, v1);
-                    Vector256<ushort> vector2 = Vector256.Equals(vector, v2);
-                    Vector256<ushort> vector3 = Vector256.Equals(vector, v3);
-                    Vector256<byte> cmp = (vector1 | vector2 | vector3).AsByte();
+                    Vector256<ushort> v1 = Vector256.Create((ushort)c);
+                    Vector256<ushort> v2 = Vector256.Create((ushort)c2);
+                    Vector256<ushort> v3 = Vector256.Create((ushort)c3);
 
-                    if (cmp != Vector256<byte>.Zero)
+                    do
                     {
-                        uint mask = cmp.ExtractMostSignificantBits() & 0x55555555u;
-                        do
+                        Vector256<ushort> vector = Vector256.LoadUnsafe(ref source, offset);
+                        Vector256<ushort> v1Eq = Vector256.Equals(vector, v1);
+                        Vector256<ushort> v2Eq = Vector256.Equals(vector, v2);
+                        Vector256<ushort> v3Eq = Vector256.Equals(vector, v3);
+                        Vector256<byte> cmp = (v1Eq | v2Eq | v3Eq).AsByte();
+
+                        if (cmp != Vector256<byte>.Zero)
                         {
-                            uint bitPos = (uint)BitOperations.TrailingZeroCount(mask) / sizeof(char);
-                            sepListBuilder.Append((int)(offset + bitPos));
-                            mask = BitOperations.ResetLowestSetBit(mask);
-                        } while (mask != 0);
-                    }
+                            uint mask = cmp.ExtractMostSignificantBits() & 0x55555555;
+                            do
+                            {
+                                uint bitPos = (uint)BitOperations.TrailingZeroCount(mask) / sizeof(char);
+                                sepListBuilder.Append((int)(offset + bitPos));
+                                mask = BitOperations.ResetLowestSetBit(mask);
+                            } while (mask != 0);
+                        }
+
+                        offset += (nuint)Vector256<ushort>.Count;
+                    } while (offset <= lengthToExamine - (nuint)Vector256<ushort>.Count);
+
+                    // See if we fit another 128 bit vector
+                    if (offset <= lengthToExamine - (nuint)Vector128<ushort>.Count)
+                    {
+                        Vector128<ushort> vector = Vector128.LoadUnsafe(ref source, offset);
+                        Vector128<ushort> v1Eq = Vector128.Equals(vector, v1.GetLower());
```

Comment:
```v1.GetLower()``` is able to reuse the same xmm used previously in Vector256.Create, so this becomes a noop.